src/gennylib/src/context.cpp

```diff
@@ -66,6 +66,12 @@ WorkloadContext::WorkloadContext(const Node& node,
             _actors.push_back(std::move(actor));
         }
     }
+
+    orchestrator.addPrePhaseStartHook([this](Orchestrator* o) {
+        for (auto&& rl : this->_rateLimiters) {
+            rl.second->resetLastEmptied();
+        }
+    });
```

Comment:
The limiters won’t change after startup so I think better to iterate outside and call addHook once for each limiter